File: captainhcg/go/350-intersection-of-two-arrays-ii.go
func intersect(nums1 []int, nums2 []int) []int {
    sort.Ints(nums1)
    sort.Ints(nums2)
    p1 := 0
    p2 := 0
    res := []int{}
    for p1 < len(nums1) && p2 < len(nums2) {
        n1 := nums1[p1]
        n2 := nums2[p2]
        if n1 == n2 {
            res = append(res, n1)
            p1 += 1
            p2 += 1
        } else if (n1 > n2) {
            p2 += 1
        } else {
            p1 += 1
        }
    }
    return res
}

File: bwu/array/350-intersection-of-two-arrays-ii.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        dict1, dict2 = self.numCount(nums1), self.numCount(nums2)
        ret= []
        for key, value in dict1.items():
            if key in dict2:
                ret += [key for _ in xrange(min(value, dict2[key]))]
        return ret
        
    def numCount(self, nums):
        dict = {}
        for n in nums:
            if n not in dict:
                dict[n] = 1
            else:
                dict[n] += 1
        return dict
